### PR TITLE
HHH-15246 fix NPE when null value is set to Enum in query

### PR DESCRIPTION
**Pull request by NathanQingyangXu**
https://hibernate.atlassian.net/browse/HHH-15246

Just another ticket about our old friend: NPE!

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/sql/ast/tree/expression/QueryLiteral.java [hunk 1]
```diff
@@ -46,23 +45,24 @@ public QueryLiteral(T value, BasicValuedMapping type) {
 				final Object literalValue = value;
 				final Object sqlLiteralValue;
 
-				if ( valueConverter.getDomainJavaType().getJavaTypeClass().isInstance( literalValue ) ) {
+				if ( literalValue == null || valueConverter.getDomainJavaType().getJavaTypeClass().isInstance(
+						literalValue ) ) {
 					sqlLiteralValue = valueConverter.toRelationalValue( literalValue );
```
Comment (NathanQingyangXu) [inline hibernate-core/src/main/java/org/hibernate/sql/ast/tree/expression/QueryLiteral.java:50]: we might need to check all the converters won't throw NPE as well

File: hibernate-core/src/main/java/org/hibernate/sql/ast/tree/expression/QueryLiteral.java [hunk 2]
```diff
@@ -46,23 +45,24 @@ public QueryLiteral(T value, BasicValuedMapping type) {
...
+					throw new SqlTreeCreationException(
+							String.format(
+									Locale.ROOT,
+									"QueryLiteral type [`%s`] did not match domain Java-type [`%s`] nor JDBC Java-type [`%s`]",
+									literalValue.getClass(),
+									valueConverter.getDomainJavaType().getJavaTypeClass().getName(),
+									valueConverter.getRelationalJavaType().getJavaTypeClass().getName()
+							)
+					);
+				}
```
Comment (NathanQingyangXu) [inline hibernate-core/src/main/java/org/hibernate/sql/ast/tree/expression/QueryLiteral.java:65]: I refactored the code to make it a little bit easier to understand.

**Comment by dreab8** [comment]
LGTM

**Comment by dreab8** [comment]
Thanks @NathanQingyangXu for the PR

**Pull request merged**